qualang_tools/octave_tools/README.md

```diff
@@ -0,0 +1,240 @@
+# Octave Tools
+This library includes tools to improve the users Octave experience.
+
+**Note that these functions only work with `qm-qua >= 1.1.5`.**
+
+## get_calibration_parameters
+This function will look into the calibration database and return the correction parameters (offsets and correction 
+matrix) corresponding to a given set of intermediate frequency, Octave LO frequency and gain. 
+
+The correction parameters are returned in the format of a dictionary:
+```
+{'offsets': {'I': I_offset, 'Q': Q_offset},
+ 'correction_matrix': (c00, c01, c10, c11)}
+```
+
+If no calibration parameters are found in the database, the function will either:
+* return the dictionary with empty values (verbose_level 0)
+* return the dictionary with empty values and print a warning in the Python console (verbose_level 1)
+* raise an error and block the execution of the script (verbose_level 2)
+
+
+### Usage example
+ 
+```python
+from qualang_tools.octave_tools import set_correction_parameters
+
+# Open the qmm and qm
+qmm = QuantumMachinesManager()
+qm = qmm.open_qm(config)
+
+# Calibrate the desired set of LO frequency and IF
+qm.octave.set_rf_output_gain("qubit", -7)
+qm.calibrate_element("qubit", {5e9: (50e6,)})
+
+# Set the new frequency with calibrated gain
+qm.octave.set_lo_frequency("qubit", 5e9)
+qm.octave.set_rf_output_gain("qubit", -7)
+# Get the correction parameters corresponding to the desired parameters:
+param_qubit = set_correction_parameters(
+    path_to_database="", config=config, element="qubit", 
+    LO=5e9, IF=50e6, gain=-7, 
+    qm=qm, verbose_level=1)
+```
+
+## set_correction_parameters
+This function will look into the calibration database and update the correction parameters in Python, (offsets and 
+correction matrix) corresponding to a given set of intermediate frequency, Octave LO frequency and gain. 
+
+The goal to give the user the ability to easily update the correction parameters while sweeping the LO frequency
+in Python for instance.
+
+The correction parameters are returned in the format of a dictionary:
+```
+{'offsets': {'I': I_offset, 'Q': Q_offset},
+ 'correction_matrix': (c00, c01, c10, c11)}
+```
```

Comment:
remove?

Reply:
Why? Here I wanted to explicitly show the structure

Reply:
This is in the set function (I agree it's needed in the get)

Reply:
Actually the set function also returns the calibration parameters in case the user wants to check it, this is why I kept it here as well

Reply:
Oh. Then it should be stated... Also, it doesn't appear in the docstring.

I understand what bothers me. Usually, you have a set and a get function to the same "object".
Here, the set and the get are for different objects: The get is from the calibration, the set is to the OPX.
That's ok, just needs to be explained. And this is actually clear in the function names and docstrings, so I wonder if we're already good?
The functions names are actually different, but "calibration" and "correction" are hard to distinguish from a quick glance...

Reply:
Yes I just stated it in the returns, I added it explicitly in the docstring now.
I wanted to make the users use only one function depending whether they want to get or set the parameters, but maybe it is just confusing...
If you think it is better I can keep the `get` function as it is and modify the `set` function to only set the parameters from what the `get` function returns as you suggested initially, what do you think?

Reply:
I'm not sure... Maybe run it with a few CS people and see what makes sense to them.
To me, having the `set` get the parameters as input is better, but it's also confusing because it updates something differently from the `get` 😆 

Maybe `get_calibration_parameters_from_db` & `set_correction_parameters_to_opx`?

Reply:
The problem is that the bandwidth is super limited because of AWS...
I like your suggestion for the names, it is way more explicit!
However, there is no straightforward way to set to correction matrix if a job is not running, right?

Reply:
Yes, I understand...
Maybe the name change is enough. We can change it to `set_correction_parameters_to_job`.
Do we only intend it to be used on a running job?

Reply:
I would prefer not to, but the only available function I could find for loading the correction matrix is from the job API...

Reply:
`qm.set_mixer_correction()`

Reply:
Yeah but you need to find the mixer name which is hidden with the Octave and maybe be subject to breaking changes...
I started to do this and then it looked to hacky for me and potentially not very robust because if you have several elements in the mixer list, then you need to make sure to modify the right one...
Do you think I should still do it?

Reply:
We can give it an element, get the config from the QM, then get the mixer name from the element.

And no, I think that doing it only for a job is ok, let's just change the function name.

Reply:
After some thinking I realized that several customers are changing the LO frequency using the API within the program and not the config and they want to update the correction parameters before running a job.
I implemented the possibility to do it, but since `qm.set_mixer_correction()` only adds a set of parameters in the `mixers` section, you also need to update the LO and intermediate frequencies of the element so that the OPX picks the right set of correction parameters.
I didn't manage to do so because the qm API doesn't allow updating the frequencies in the loaded config (without reopening a QM) so I just updated the correction matrix but not the corresponding frequencies (this is what I called hacky above). For instance:
`{'intermediate_frequency': 50e6, 'lo_frequency': 4e9, 'correction': old_matrix)}`
`set_correction_parameters_to_opx(if=-10e6, lo=6e9)`
`{'intermediate_frequency': 50000000.0, 'lo_frequency': 4000000000.0, 'correction': new_matrix)}`
What do you think?

Reply:
I think that this is the correct solution for now.
In the (near) future, the correction parameters will not be part of the QM, but part of the job.

Reply:
What about changing the names of the functions?
`get_calibration_parameters_from_db` & `set_correction_parameters_to_opx`

Reply:
I already updated the names, do you mean changing these to something else?

Reply:
Oh, I missed that.
Please update it in the changelog as well!

Reply:
Good catch, done!